cli/cmd/proxy/list/command.go

```
@@ -0,0 +1,228 @@
+
+	// Parse the command line flags.
+	if err := c.set.Parse(args); err != nil {
+		c.UI.Output("Error parsing arguments: %v", err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Validate the command line flags.
+	if err := c.validateFlags(); err != nil {
+		c.UI.Output("Invalid argument: %v", err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	if c.kubernetes == nil {
+		if err := c.initKubernetes(); err != nil {
+			c.UI.Output("Error initializing Kubernetes client", err.Error(), terminal.WithErrorStyle())
+			return 1
+		}
+	}
+
+	pods, err := c.fetchPods()
+	if err != nil {
+		c.UI.Output("Error fetching pods:", err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	c.output(pods)
+	return 0
+}
+
+// Help returns a description of the command and how it is used.
+func (c *ListCommand) Help() string {
+	c.once.Do(c.init)
+	return c.Synopsis() + "\n\nUsage: consul-k8s proxy list [flags]\n\n" + c.help
+}
+
+// Synopsis returns a one-line command summary.
+func (c *ListCommand) Synopsis() string {
+	return "List all Pods running proxies managed by Consul."
+}
+
+// validateFlags ensures that the flags passed in by the can be used.
+func (c *ListCommand) validateFlags() error {
+	if len(c.set.Args()) > 0 {
+		return errors.New("should have no non-flag arguments")
+	}
+	if errs := validation.ValidateNamespaceName(c.flagNamespace, false); c.flagNamespace != "" && len(errs) > 0 {
+		return fmt.Errorf("invalid namespace name passed for -namespace/-n: %v", strings.Join(errs, "; "))
+	}
+	return nil
+}
+
+// initKubernetes initializes the Kubernetes client and sets the namespace based
+// on the user-provided arguments.
+func (c *ListCommand) initKubernetes() error {
+	settings := helmCLI.New()
+
+	restConfig, err := settings.RESTClientGetter().ToRESTConfig()
+	if err != nil {
+		return fmt.Errorf("error retrieving Kubernetes authentication %v", err)
+	}
+	c.kubernetes, err = kubernetes.NewForConfig(restConfig)
```

error check here and fail fast?